### PR TITLE
Adds a tea plant and matcha!

### PR DESCRIPTION
**Pull request by avimour**
## About the PR 
Adds in a new botany plant, tea! You can grow tea and use it's leaves to create matcha, either by processing it in a processor or distilling it with a still. Matcha can then be used to create matcha tea, which is known to reduce cholesterol in someone's body.
The plant itself contains tea for some interesting botany splices!

image image

I do have a plan for more things for both matcha and the tea plant itself, however I'll add that in another PR since I don't want this one to get too big.

## Why's this needed? 
More variety is nice! Matcha tea can be used to purge cholesterol from someone which can be nice for a bartender to make for when the chef starts making unhealthy foods. Plus gives the chef an option for tasty green icing! With the tea plant containing tea you can also get some interesting splices in botany now, such as a half and half berry!

## Changelog

```changelog
(u)avimour
(*)Added a new botany plant: Tea. Which can be used to create matcha!
(+)Combine matcha and water to get...matcha tea!
```

### CODE REVIEW

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 1]
```diff
@@ -4365,3 +4365,35 @@ datum
...
+			energy_value = 0.04
+			addiction_prob = 1
+			addiction_prob2 = 2
+			addiction_min = 10
+			var/list/flushed_reagents = list("cholesterol")
+
+			on_mob_life(var/mob/M, var/mult = 1)
+				flush(M, 3 * mult, flushed_reagents)
+				..()
+				return
```
Comment (TobleroneSwordfish) [inline code/modules/chemistry/Reagents-FoodDrink.dm:4399]:
```suggestion
```

File: code/modules/food_and_drink/condiments.dm [hunk 2]
```diff
@@ -106,6 +106,14 @@
 	bites_left = 2
 	heal_amt = 3
 
+/obj/item/reagent_containers/food/snacks/condiment/matcha
+	name = "matcha"
+	desc = "A powder created from dried tea leaves that could help with."
```
Comment (TobleroneSwordfish) [inline code/modules/food_and_drink/condiments.dm:111]: Could help with?

File: code/modules/hydroponics/plants_herb.dm [hunk 3]
```diff
@@ -226,6 +226,21 @@ ABSTRACT_TYPE(/datum/plant/herb)
...
+/datum/plant/herb/tea
+	name = "Tea"
+	seedcolor = "#377a41"
+	crop = /obj/item/plant/herb/tea
+	starthealth = 20
+	growtime = 20
+	harvtime = 60
+	cropsize = 5
+	harvests = 1
+	isgrass = 1
```
Comment (TobleroneSwordfish) [inline code/modules/hydroponics/plants_herb.dm:238]:
```suggestion
	isgrass = TRUE
```

File: code/obj/item/plants.dm [hunk 4]
```diff
@@ -351,6 +351,14 @@
 	desc = "A distinctive red flower."
 	icon_state = "poppy"
 
+/obj/item/plant/herb/tea
+	name = "tea leaves"
+	crop_suffix	= " leaves"
```
Comment (TobleroneSwordfish) [inline code/obj/item/plants.dm:356]:
```suggestion
	crop_suffix = " leaves"
```

File: code/obj/item/plants.dm [hunk 5]
```diff
@@ -351,6 +351,14 @@
...
+	desc = "Leaves from a green tea plant, which can be used to create matcha."
+	icon_state = "tealeaves"
+	brewable = 1
```
Comment (TobleroneSwordfish) [inline code/obj/item/plants.dm:359]:
```suggestion
	brewable = TRUE
```

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
I'm amazed we don't already have a tea plant, this looks great

File: code/modules/food_and_drink/condiments.dm (hunk 2 above)
Comment (avimour) [reply code/modules/food_and_drink/condiments.dm:111]: AAA i forgot to finish the sentence oTL

**Review by WALPVRGIS — APPROVED** [review]
Very cute, glad to see these made! Lovely artwork. :)

**Comment by Tarmunora** [comment]
Does nobody else think it's a little weird that you brew/process the tea leaves into matcha, which then has to be mixed with tea *extracted* from tea leaves in order to make matcha tea? Because I find that process a little weird

**Comment by WALPVRGIS** [comment]
I don't think it's too weird as a concept for the game, though as an analogue to irl matcha recipes is a bit odd upon some thought sdgjhf. Maybe just matcha and water would make more sense? My approval is based on sprites btw, leaving the chem stuff to you tarm aha. 👍

**Comment by avimour** [comment]
!merge_upstream

**Comment by Tarmunora** [comment]
🤷 I still think the way it's arranged is a *little* weird, but not a dealbreaker or anything

**Pull request merged**